### PR TITLE
Replace Pester tests with xunit tests

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
This replaces the Pester E2E tests with xunit E2E tests and uses all Omnisharp types (no reliance on PSES.Protocol!)

The biggest problem I had with the Pester tests was that it was so hard to debug them... and the programming model relied too much on timing things perfectly.

I've moved over to xUnit tests as the asynchronous nature of communicating with PSES proved to be really tough with PowerShell.

I _think_ this new model is a little more stable and uses Omnisharp's client so the types are consistant with no reliance on our protocol project.

The tests do exactly what the Pester tests do, just using xunit instead.

This switches to starting PSES via `Stdio` testing that code path... It might be worthwhile to have a `NamedPipe` smoke test (that just connects and disconnects)... but that's out of scope for now. 

Also, that might have to use Pester because I was experiencing some hanging on Windows getting the NamedPipes working with xUnit.

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Now the tests don't use a REPL which is probably ideal for our testing environment

**Comment by TylerLeonhardt** [comment]
oops - I tried adding named pipe tests and no dice. Moving to stdio

File: test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+                }
+            });
+
+            return filePath;
+        }
+
+        private async Task WaitForDiagnostics()
+        {
+            // Wait for PSSA to finish.
+            var i = 0;
```
Comment (rjmholt) [inline test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs]:
```suggestion
            int i = 0;
```

File: test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+            }
+        }
+
+        [Fact]
+        public async Task CanSendPowerShellGetVersionRequest()
+        {
+            PowerShellVersionDetails details
+                = await LanguageClient.SendRequest<PowerShellVersionDetails>("powerShell/getVersion", new GetVersionParams());
+
+            if(Environment.GetEnvironmentVariable("PWSH_EXE_NAME") == "powershell")
```
Comment (rjmholt) [inline test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs]: Worth making this a static?

File: test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+                {
+                    Range range = location2.Range;
+                    Assert.Equal(5, range.Start.Line);
+                    Assert.Equal(0, range.Start.Character);
+                    Assert.Equal(5, range.End.Line);
+                    Assert.Equal(24, range.End.Character);
+                });
+        }
+
+        [Fact(Skip = "Bug in Newtonsoft that csharp-language-server-protocol will workaround")]
```
Comment (rjmholt) [inline test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs]: Possible to explain what the issue is and/or link to the issue here in a comment?

File: test/PowerShellEditorServices.Test.E2E/TestsFixture.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                "-FeatureFlags", string.Join(',', s_featureFlags),
+                "-HostName", s_hostName,
+                "-HostProfileId", s_hostProfileId,
+                "-HostVersion", s_hostVersion,
+                "-AdditionalModules", string.Join(',', s_additionalModules),
+                "-BundledModulesPath", s_bundledModulePath,
+                "-Stdio"
+            };
+
+            var base64Str = Convert.ToBase64String(
```
Comment (rjmholt) [inline test/PowerShellEditorServices.Test.E2E/TestsFixture.cs]:
```suggestion
            string base64Str = Convert.ToBase64String(
```

File: test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs (hunk 3 above)
Comment (TylerLeonhardt) [reply test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs]: They've fixed the issue :) and the tests are passing now!

File: test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply test/PowerShellEditorServices.Test.E2E/LanguageServerProtocolMessageTests.cs]: done

**Pull request merged**